### PR TITLE
Update phan/phan and drenso/phan-extensions to latest

### PR DESCRIPTION
**Pull request by dairbuirabass**
## Description
Update phan/phan to latest (5.3.0) and drenso/phan-extensions to latest (3.5.1)

## Related Issue
-

## Motivation and Context
Keeping dev packages up to date

## How Has This Been Tested?
Locally

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [ ] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Plugin/Catalog/Model/Config.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ public function afterGetAttributesUsedForSortBy(// phpcs:ignore Generic.CodeAnal
 
             $eavAttribute = $this->attributeFactory->createAttribute(Attribute::class);
             $eavAttribute->setAttributeCode(NostoHelperSorting::NOSTO_PERSONALIZED_KEY);
-            //@phan-suppress-next-line PhanTypeMismatchArgument
+            /** @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal */
```
Comment (mridang) [inline Plugin/Catalog/Model/Config.php:129]: Why is this? Could you add a comment please.

File: Model/Service/Facet/BuildWebFacetService.php [hunk 2]
```diff
@@ -229,7 +229,7 @@ private function mapIncludeFilter(Store $store, IncludeFilters &$includeFilters,
      */
     private function getCategoryName(Store $store, $categoryId): ?string
     {
-        //@phan-suppress-next-next-line PhanTypeMismatchArgument
+        /**  @phan-suppress-next-next-line PhanTypeMismatchArgumentSuperType */
```
Comment (mridang) [inline Model/Service/Facet/BuildWebFacetService.php:232]: Why is this? Could you add a comment please.

File: Helper/CategorySorting.php [hunk 3]
```diff
@@ -78,6 +78,7 @@ public function __construct(
     public static function getNostoSortingOptions()
     {
         return [
+            /** @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal */
```
Comment (mridang) [inline Helper/CategorySorting.php:81]: Why is this? Could you add a comment please.

**Review by mridang — APPROVED** [review]

**Comment by mridang** [comment]
Update the title please

File: Helper/CategorySorting.php (hunk 3 above)
Comment (dairbuirabass) [reply Helper/CategorySorting.php:81]: The functions `__()` expects array as a parameter in phpDoc: https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Phrase/__.php#L16

But it supports passing the string and that's how they use it throughout, so I think phpDoc could be improved from M2 side

**Pull request merged**